plugins/fmeflow-common/src/index.ts

```diff
@@ -1,37 +1,115 @@
+// packages/fmeflow-common/src/index.ts
+
 import { Entity, KindValidator } from '@backstage/catalog-model';
 import { JsonObject } from '@backstage/types';
 
-/**
- * FMEWorkspace Entity Definition
- */
+interface FMELogEntry  {
+  line: string;
+  message: string;
+}
+
 export interface FMEWorkspaceEntity extends Entity {
   apiVersion: 'geoportia.se/v1alpha1';
   kind: 'FMEWorkspace';
   spec: {
     type: string;
-    lifecycle: string;
-    owner: string;
+    lifecycle?: string;
+    owner?: string;
     lastUpdated?: string;
   } & JsonObject;
 }
 
-/**
- * Type guard to check if an Entity is an FMEWorkspaceEntity
- */
 export const isFMEWorkspaceEntity = (entity: Entity): entity is FMEWorkspaceEntity =>
   entity.apiVersion === 'geoportia.se/v1alpha1' &&
   entity.kind === 'FMEWorkspace' &&
   typeof entity.spec === 'object' &&
-  typeof entity.spec.owner === 'string' &&
-  typeof entity.spec.type === 'string' &&
-  typeof entity.spec.lastUpdated === 'string' &&
-  typeof entity.spec.lifecycle === 'string';
-
-/**
- * Validator to be registered with the catalog for FMEWorkspace entities
- */
-export const fmeWorkspaceEntityValidator: KindValidator = {
-  async check(entity: Entity): Promise<boolean> {
-    return isFMEWorkspaceEntity(entity);
-  },
-};
+  typeof entity.spec.type === 'string'
+
+
+export const isResourceEntity = (entity: Entity): boolean =>
+    entity.apiVersion === 'geoportia.se/v1alpha1' &&
+    entity.kind === 'Resource' &&
+    typeof entity.metadata?.name === 'string';
+  
+export const isTableEntity = (entity: Entity): boolean =>
```

Comment:
Should use the existing function https://github.com/Eskilstuna-kommun/geoportia/blob/dev/plugins/postgresql-data-common/src/index.ts#L26

Though it should probably be pulled from the postgres-specific package, maybe introducing a `database-common` package with the basic Table entity definition and function, adding a `dialect: "postgres"` attribute to the entity. The code here would then use `isTableEntity` from `database-common` which only checks apiVersion and kind, while postgres-specific code would use `isPostgresTableEntity` which additionally checks `dialect`?